### PR TITLE
Enable showing border shields on uninterruptible casts

### PR DESCRIPTION
**Pull request by alexzaworski**
Hi,

First of all thank you very much for this addon, Classic is borderline unplayable without it. Regrettably I am quite dumb and often try to interrupt physical casts such as war stomp and grenades :(

This PR does the following:
- Adds a (small, very incomplete) dictionary of uninterruptible casts. This list is based on things I encounter often in pvp and can be expanded substantially over time.
- Shows a border shield when one of those spells is being cast
- Recolors the cast bar when one of those spells is being cast (to the default uninterruptible color per Blizzard client code)
- Adds an option to show/hide the shields per-castbar (off by default)
- Adds an option to change the uninterruptible color per-castbar
- Adds an option to control the icon offset while the border shield is active, set to sensible defaults

This PR _does not_ do the following:
- Gracefully handle border shields with skinned cast bars. Not sure how to do that, unfortunately
- Localize the added options

Other notes
- The border shield looks kinda wonky on nameplates, presumably since they were made to work with Retail nameplates
- I hardcoded some text offsets just to get things up and running, if you'd like them migrated to a setting I am happy to do so.
- Hiding/showing the border shields should maybe be a global setting, dunno
- I've not written much lua nor have I actually done any real addon dev so there may be some silly mistakes here

### CODE REVIEW

File: ClassicCastbars/core/Data.lua [hunk 1]
```diff
@@ -1569,9 +1569,32 @@ namespace.pushbackBlacklist = {
...
+    [GetSpellInfo(13808)] = 1,      -- M73 Frag Grenade
+    [GetSpellInfo(4069)] = 1,       -- Big Iron Bomb
+    [GetSpellInfo(12543)] = 1,      -- Hi-Explosive Bomb
+    [GetSpellInfo(4064)] = 1,       -- Rough Copper Bomb
+    [GetSpellInfo(12421)] = 1,      -- Mithril Frag Bomb
+    [GetSpellInfo(19784)] = 1,      -- Dark Iron Bomb
+    [GetSpellInfo(4067)] = 1,       -- Big Bronze Bomb
+    [GetSpellInfo(4066)] = 1,       -- Small Bronze Bomb
+    [GetSpellInfo(4065)] = 1,       -- Large Copper Bomb
+    [GetSpellInfo(13237)] = 1,      -- Goblin Mortar
```
Comment (alexzaworski) [inline ClassicCastbars/core/Data.lua]: Mortar doesn't have a cast time 😄

**Comment by wardz** [comment]
Added some WIP changes that should fix most of the issues you mentioned. Will comment further on it later.

**Comment by alexzaworski** [comment]
Thanks so much for picking this up! I haven't played with your patch yet but a lot of the changes look great— auto-detecting uninterruptible stuff for NPCs is awesome.

File: ClassicCastbars/core/Data.lua (hunk 1 above)
Comment (wardz) [reply ClassicCastbars/core/Data.lua]: Thanks, removed it now.

**Comment by wardz** [comment]
Merging now as it seems to work good enough. There's still some extra features i'd like to add though, so it might take some time before it makes it to release.

**Pull request merged**